README.md

```diff
@@ -1,66 +1,112 @@
-| Clippy (Rust linter)     | [`.clippy.toml`](./.clippy.toml)                         | [Repository](https://github.com/rust-lang/rust-clippy), [ Reference ]( https://rust-lang.github.io/rust-clippy/) |
-| Rustfmt (Rust formatter) | [`.rustfmt.toml`](./.rustfmt.toml)                       | [Repository](https://github.com/rust-lang/rustfmt), [ Reference](https://rust-lang.github.io/rustfmt/)           |
-| Commitlint               | [`.commitlintrc.json`](./.commitlintrc.json)            | [Repository](https://github.com/conventional-changelog/commitlint), [Reference](https://commitlint.js.org/#/)    |
-| `cargo-deny`             | [`deny.toml`](./deny.toml)                               | [Repository](https://github.com/EmbarkStudios/cargo-deny)                                                        |
-
-## Run scripts locally
- - Run unit/integration/doc tests: `cargo test`
- - Run fuzz tests: `cargo fuzz <fuzz-target>`
- - Run Rustfmt: `cargo fmt`
- - Run Clippy: `cargo clippy`
- - Run performance benchmarks: `cargo bench`
- - Generate API docs for crate: `cargo doc`
- - Generate mdBook docs for crate: `mdbook build`
- - Run security audits: `cargo audit` (requires installing [`cargo-audit`](https://crates.io/crates/cargo-audit) locally)
- - Lint commit messages: `npm run lint:commit` (requires installing [`commitlint`](https://commitlint.js.org/#/) locally with `npm install`)
-
-----
-
-# {{library}}
-{{library description}}
+# Differ.rs
+This crate provides edit distance, delta vectors between 2 words, and lets you apply delta vectors in order to transform words.
 
 ## Install
 ```shell
-cargo add {{library}}
+cargo add differ-rs
+```
+or, simply add the following string to your Cargo.toml:
+```
+differ-rs = "0.0.0"
+```
+
+## Features
+* `apply_diff` : Allows users to apply delta vectors in order to transform a words.
+* `extra_traits`: all `struct`s implemented in `differ-rs` are `HammingDistance` and `LevenshteinDistance`. Each Struct implements the `diff` and `distance` methods. 
+
+## How it works
+* `apply_diff`works by giving a string and a transformation vector to the method. Then the transformation vector is applied to the string given in the first argument.
+* `StringDiffAlgorithm` provides two methods `diff` which gives you a transformation vector from the first to second string. the `distance` method gives you the edit distance from the frist argument to the second argument. The structs `HammingDistance` and `LevenshteinDistance` have their own implementations for each method.
+
+## Examples
+
+Getting the edit distance between two words using Levenshtein algorithm 
+```
+use differ_rs::{LevenshteinDistance, StringDiffAlgorithm};
+fn main(){
+    let my_levensthein = LevenshteinDistance {};
+
+    let edit_distance = my_levensthein.distance("Sitting", "Kitten");
+    
+    assert_eq!(3, edit_distance)
+}
+```
+Note: We are getting the edit distance to get from "Sitting" to "Kitten".
```

Comment:
Can you replace this with specially-formatted blockquotes? It'll render as:

> **Note**: We are getting the edit distance to get from "Sitting" to "Kitten".

You can do this by changing "Note: [...]" to "> **Note**: [...]"